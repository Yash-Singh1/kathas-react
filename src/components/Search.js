import React, { Component } from 'react';
import NavigationBar from './NavigationBar';
import textBoundary from '../helpers/textBoundary';
import { Link } from 'react-router-dom';
import pagesAndContent from '../data/text';

class Search extends Component {
  render() {
    const query = decodeURIComponent(
      new URLSearchParams(this.props.location?.search).get('q')
    );

    const filteredResults = pagesAndContent.filter((content) =>
      content.text.includes(query)
    );

    return (
      <NavigationBar search>
        {filteredResults.length === 0 ? (
          <p>No results found</p>
        ) : (
          filteredResults.map((pageAndContent, id) => (
            <div key={id} style={{ textAlign: 'left' }}>
              <p>
                <Link to={pageAndContent.path}>{pageAndContent.name}</Link>
              </p>
              <p>
                {pageAndContent.text.split(query).map((v, index) => (
                  <React.Fragment key={index}>
                    {textBoundary(pageAndContent.text, query, index + 1)}
                    <br />
                  </React.Fragment>
                ))}
                {textBoundary(pageAndContent.text, query)}
              </p>
            </div>
          ))
        )}
      </NavigationBar>
    );
  }
}

export default Search;
