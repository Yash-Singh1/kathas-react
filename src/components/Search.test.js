import React from 'react';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {
  var search = shallow(<Search />);

  it('renders properly', () => {
    expect(search.instance()).toMatchSnapshot();
  });

  it('contains the navbar', () => {
    expect(search.find('NavigationBar').exists()).toBe(true);
  });

  it('shows results', () => {
    search = shallow(<Search location={{ search: '?q=About' }} />);
    expect(search.find('NavigationBar').children().length).toEqual(1);
  });
});
