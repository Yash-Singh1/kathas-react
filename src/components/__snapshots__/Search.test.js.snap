// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders properly 1`] = `
Search {
  "context": Object {},
  "props": Object {},
  "refs": Object {},
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useDeferredValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useResponder": [Function],
        "useState": [Function],
        "useTransition": [Function],
      },
      "_element": <Search />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <NavigationBar
        search={true}
      >
        <p>
          No results found
        </p>
      </NavigationBar>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
